samples/drivers/adc/src/main.c

```diff
@@ -51,29 +58,35 @@ void main(void)
 			printk("Could not setup channel #%d (%d)\n", i, err);
 			return;
 		}
+
+		/* Store separately because it's overwritten by `adc_sequence_init_dt`*/
+		channels |= BIT(adc_channels[i].channel_id);
 	}
 
+	/* The sequence settings will only depend on the first channel.
+	 * Any settings on other channels will be ignored.
+	 */
+	(void)adc_sequence_init_dt(&adc_channels[0], &sequence);
+
+	sequence.channels = channels;
```

Comment:
This would essentially change the behavior of the sample and it would actually break the sample for the `nrf52840dk_nrf52840` board which presents usage of [different resolution and oversampling settings for particular channels](https://github.com/zephyrproject-rtos/zephyr/blob/10130d8e6de8c3889f42d498d3de6115db8f2d35/samples/drivers/adc/boards/nrf52840dk_nrf52840.overlay#L18-L35), what is also [mentioned in the sample description](https://github.com/zephyrproject-rtos/zephyr/blob/10130d8e6de8c3889f42d498d3de6115db8f2d35/samples/drivers/adc/README.rst#L29-L34).

I think it's better to add a separate sample if you want to showcase multichannel sequences.

Reply:
I don't mind simply not supplying a sample. I only added the changes because I used it during development. So would you mind if we simply drop any changes to `samples/`?

Reply:
Sure, let's just drop those changes.

Reply:
@heinwessels What about adding a new multiple channel sample ?

Reply:
@erwango if a sample is not required then I'd rather continue on my other Zephyr PRs that's still broken ;). There's also not that big of a difference between sampling a single channel and multiple channels.

Reply:
Changes to `samples/drivers/adc/` removed by dropping that commit.

Reply:
It was good value though, it's just that it would not run on all boards so it should not be the only sample provided. I think it'd be cool if you feel like bringing it back as a separate test in a followup PR since you had the work done already, just modified with a filter and config that is known to work that people can add others.

Reply:
I've had to explain how multi-channel sampling works more than once.  +1 to bringing the example back in another PR.